src/permission/Permission.php

```diff
@@ -36,15 +36,17 @@ class Permission{
 	private Translatable|string $description;
 
 	/**
-	 * Creates a new Permission object to be attached to Permissible objects
-	 *
-	 * @param bool[] $children
+	 * @var bool[] $children
 	 * @phpstan-param array<string, bool> $children
 	 */
+	private array $children = [];
+
+	/**
+	 * Creates a new Permission object to be attached to Permissible objects
+	 */
 	public function __construct(
 		private string $name,
 		Translatable|string|null $description = null,
-		private array $children = []
```

Comment:
Not clear why this is removed

Reply:
It's remove because currently array signature is `array<string, bool>` but now with Permission we can't put `array<Permission, bool>` so i remove this array and use `$this->addChild()` at replacement

Reply:
Removing the ability to define children in the constructor may prove more complex. Could we imagine keeping this and calling addChild? Or make addChild / removeChild chainable by returning them themselves.